### ISSUE TITLE
HUD Rework release - what you need to know

### ISSUE THREAD

**Issue by Tensuko** [Courseplay/courseplay#3552]
Hallo together,
the new HUD is ready to be merged into the actual CP Version.
Before you get into panic, I give you some Informations about what has changed:
- You will only be able to get into a Menu or a Mode, when the Device you are in/attached are supported.
- The old Coursegenerator got removed and you will find the new Coursegenerator at the 2nd Page (see Pics).
- There are no new Modes activated yet, as we still need to rework the Mode2 (what comes next).
- New Feature: A mini-HUD, that shows you some information without activated HUD (you can disable this in the settings)
- Some Settings changed the Menu, wich are more logical then before (more clean menus).

With the new HUD, we got a lot more freedom to change and add stuff to the HUD.

The Pics are in german Version (sry), but I will try to explain that a bit.

**Comment by Tensuko** [comment]
The Button to open the new Coursegenerator is located here:
![KursGen](https://user-images.githubusercontent.com/27344744/55781435-6bbd0280-5aaa-11e9-9e72-9961936a958b.png)
That is the Mini-HUD, it will display more when you change the Mode, CP is running and depending on the task, it will show you stuff like transporting grain run X/6 for example. It can be turned off if you don't want to see it.
![MiniHUD](https://user-images.githubusercontent.com/27344744/55781438-6cee2f80-5aaa-11e9-8b34-9cbb959d3e82.png)
This is a preview of how the CP Menus look like, when switch the Devices that are attached.
![Vehicle](https://user-images.githubusercontent.com/27344744/55781443-6f508980-5aaa-11e9-9db6-3d23cea9e400.png)

We hope you like this, any questions, feel free to ask.
If you face any real Issue, start one and discripe whats wrong, we will take a look into this.

**Comment by Messy71** [comment]
this is in commit [6858f6eb](https://github.com/Courseplay/courseplay/commit/6858f6eb1a4f2114ceacb14ef11392e3b2c01886) in HUD_Refactoring branch and not in the master branch, right?

**Comment by ThomasGaertner** [comment]
HUD_Refactoring is the branch related,  yes

**Comment by pvaiko** [comment]
The course generator should be just "Kursgenerator...." or "Course generator...", without Advanced or Erweiterter and put ... (ellipsis) behind it to show that it opens another window. Please.

**Comment by ThomasGaertner** [comment]
its online now, v.168

**Comment by hoefti10** [comment]
I cant make the choise wich type of 
fertilizer i will use. I'm blind?!? :-)
The normale course generator is replaced with the advanced ? No possibility to use the normal one?

**Comment by Tensuko** [comment]
The normal one is outdated.
The 'advanced' is just the new better one.

**Comment by pops64** [comment]
Mode2 is missing turn on field. Drive Now button is visible in all cases. Should only be shown in particular cases that it actually does something.

**Comment by Williwillswisse** [comment]
Mode 6 is missing for collecting straw.
![K1600_Pic](https://user-images.githubusercontent.com/47098121/55837334-5e8f2a80-5b21-11e9-988b-327ccd414866.JPG)

**Comment by Tensuko** [comment]
Can you pls create an own Issue for stuff like this ?

**Comment by AgroWillem** [comment]
Looks very nice. I like the new integration in FS.

**Comment by hoefti10** [comment]
> The normal one is outdated.
> The 'advanced' is just the new better one.

Oh, okay... I miss the Automatic width by the 'New One'

**Comment by pvaiko** [comment]
@hoefti10  Just pushed auto width this morning...

**Comment by XPiRiJenZ** [comment]
tbh the reworked HUD is nice, it's freat that you only enabled or disabled the modes which weren't for the attached tool. that makes cp-usage a lot easierer for ppl, who are not working with cp so far.
well... there is a small "but": why did you removed the simple/old course generator?`any cahnces to bring it back? using the advanced generator takes too long (yes, it looks fancy), but a more simple version would be nice or a setting to choose which one you prefer more...

**Comment by Tensuko** [comment]
The old Coursegenerator, was outdated and missed some options.
We won't bring it back.
The new one got some rly nice features and the advantage, that you can very easy see, what you did there and if you don't like it, change the option and just press generate again.

If we would create a simple create button, it would directly create a course with the calculated work with, up/down rows and current position. Not all Fields support this.

And I must say, I was never in rush to create a course. And if you want to create one fast, open the generator, select field and press create. It is almost same steps...

**Comment by Peppobrasil** [comment]
sometimes the HUD disappears.

**Comment by Tensuko** [comment]
@Peppobrasil 
Create a real Issue with this and post your logfile with debug channel 18 on and reproduce the error and discripe.

**Comment by DemonRaven** [comment]
I have anymore drive now button on silo to sale point mode, so my worker just sits and waits.

**Comment by Peppobrasil** [comment]
> @Peppobrasil
> Create a real Issue with this and post your logfile with debug channel 18 on and reproduce the error and discripe.

[log.txt](https://github.com/Courseplay/courseplay/files/3069225/log.txt)
![fsScreen_2019_04_11_16_44_37](https://user-images.githubusercontent.com/25421559/55967225-31a25b00-5c7a-11e9-8826-b31480d6fc2f.png)

**Comment by Peppobrasil** [comment]
How can I create a path with x spirals plus the normal up / down with the new HUD?

**Comment by Tensuko** [comment]
Not sure what you mean exactly, but take a look in the 'new' Coursegenerator and try out the settings. Maye you find what you need.

**Comment by Peppobrasil** [comment]
I tried without success. Something like this. I use it when there is no space along the field's borders
<img width="960" alt="fsScreen_2019_04_13_09_24_36" src="https://user-images.githubusercontent.com/25421559/56076235-ecdc0880-5dce-11e9-9b27-800a658dc8d5.png">

**Comment by Messy71** [comment]
I really miss the hovering over tooltips in the bottom left corner, are they going to come back anytime soon ?

**Comment by Tensuko** [comment]
@Peppobrasil 
But that looks like the normal Headland setting. You find that in the top right corner of the new coursgenerator. Just keep try out the settings and click on generate. You will find it out very easy ;)

@Messy71 
and what exaclty ? The Debugchannels got informations.
On what point you need Informations, that options are not clear enough ? :)

**Comment by PavementPilot** [comment]
Nice rework of the HUD, but I miss the field calculation for amount of seed needed for the current course. Will that be coming back?

**Comment by Tensuko** [comment]
It is in Issues as suggestion #3224

**Comment by cwattyeso** [comment]
Is there currently anyway to remove that 'No Course Loaded' message that displays all the time on screen since the update to Courseplay? I don't know why it would need to be displayed at all on screen. Just adds something which is right smack in the middle of your eyeline and ruins cleanness of the screen.
![2019-04-15](https://user-images.githubusercontent.com/13921716/56148443-3597eb00-5fa2-11e9-98fd-5d06301adaf3.png)

**Comment by Tensuko** [comment]
If you would read this Issue from the start, you would now.

**Comment by cwattyeso** [comment]
I've tried to read this page, but information is quite lacking on the new options and changes in this update, also the pictures that have been posted are all in a foreign language to me which I don't read or understand. Would be nice to get some English pictures in future to help the majority of people who don't speak foreign.

**Comment by Tensuko** [comment]
> That is the Mini-HUD, it will display more when you change the Mode, CP is running and depending on the task, it will show you stuff like transporting grain run X/6 for example. It can be turned off if you don't want to see it.
> ![MiniHUD](https://user-images.githubusercontent.com/27344744/55781438-6cee2f80-5aaa-11e9-8b34-9cbb959d3e82.png)
> This is a preview of how the CP Menus look like, when switch the Devices that are attached.
> ![Vehicle](https://user-images.githubusercontent.com/27344744/55781443-6f508980-5aaa-11e9-9db6-3d23cea9e400.png)

...

**Comment by cwattyeso** [comment]
Okay yes I found that option and Deactivated it, however the very next vehicle I went to the bloody 'No Course Loaded' thing appeared again on screen. Does it have to be turned off on every single vehicle and pieces of equipment or is there another setting somewhere just to disable the new Mini-Hud altogether across the whole mod? I don't really want additional huds and stuff on my screen, especially ones that will pop up anytime I buy or lease a new piece of equipment. In game anytime I want to know what Courseplay is doing or trying to do I'll press the right mouse button and open the menu.
